### DISCUSSION TITLE
Problem with grain recalculation after filling to grain boundaries

### DISCUSSION THREAD

**Discussion by JasonOttUW** [mtex-toolbox/mtex#2030] [Ask Anything] [answered]
Hello, everyone,

I have been using this snippet of code to calculate grain boundaries and subgrain boundaries, followed by filling to those boundaries and recalculating the grains a second time to compute the mis2mean again on the filled data. I have had no problems with this previously, but on this most recent map, I am getting an error when trying to compute the mis2mean on the filled data. I am using the commands:
```
[grains,ebsd.grainId,ebsd.mis2mean]=calcGrains(ebsd,'threshold',[2*degree 10*degree]);
ebsdFilled = fill(ebsd,grains);
[grains,ebsdFilled.grainId,ebsdFilled.mis2mean]=calcGrains(ebsdFilled,'threshold',[2*degree 10*degree]);

and the related error messages are:

_Index exceeds the number of array elements. Index must not exceed 4981.

Error in quaternion/accumarray (line 53)
  omega = real(acos(flip./s(subs) .* ...

Error in EBSD/calcGrains (line 140)
  [meanRotation, GOS] = accumarray(grainId(:),q(:),'robust');

Error in CY107_2_WBV (line 75)
[grains,ebsdFilled.grainId,ebsdFilled.mis2mean]=calcGrains(ebsdFilled,'threshold',[2*degree 10*degree]);_
```
Does anyone have any suggestions on what might be causing this mismatch with the number of array elements when I call the calcGrains the second time? 

Thanks in advance for any suggestions.

**Comment by kilir** [comment, marked as answer]
Hi Jason,

only a workaround but you can try to switch to the "old" method and see if this works for a start.

Replace the 0 with a 1:
https://github.com/mtex-toolbox/mtex/blob/72a4796aa5f5dc92dc6b6dcd56faf3130ded94ad/EBSDAnalysis/%40EBSD/calcGrains.m#L139

Does that help?

Cheers,
Rüdiger

**Reply by JasonOttUW** [reply]
Thanks for the workaround-that prevents the error. Much appreciated!!
